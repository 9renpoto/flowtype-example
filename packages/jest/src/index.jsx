/* @flow */
import React, { Component } from 'react'

export class App extends Component<{}, void> {
  render () {
    return <div>{'success'}</div>
  }
}

export class AppFailedOver extends Component<{}, void> {
  constructor (props) {
    super(props)
    this.state = { hasError: false }
  }

  componentDidCatch (_, info) {
    this.setState({ hasError: true })
  }

  render () {
    if (this.state.hasError) {
      return <h1>Something went wrong.</h1>
    }
    return this.props.children
  }
}
