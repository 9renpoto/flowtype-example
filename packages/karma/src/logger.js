/* @flow */

export function print () {
  console.log(`welcome logger`)
}
