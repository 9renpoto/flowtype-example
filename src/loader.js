const script = document.createElement('script')
const filename = Math.random() > 0.5 ? 'webpack' : 'browserify'
script.src = `https://9renpoto.github.io/js/${filename}.bundle.js`
document.body.appendChild(script)
