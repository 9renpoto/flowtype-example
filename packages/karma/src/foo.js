/* @flow */
import { print } from './logger'
print()

export const sum = function (a: number, b: number): number {
  return a + b
}
