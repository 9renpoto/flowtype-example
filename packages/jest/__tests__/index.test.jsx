/* @flow */
import React from 'react'
import { create } from 'react-test-renderer'
import { App } from '../src'

describe('React', () => {
  test('Failed', () => {
    const tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot()
  })
})
