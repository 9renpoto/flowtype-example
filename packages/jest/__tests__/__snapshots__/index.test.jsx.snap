// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React Failed 1`] = `
<div>
  success
</div>
`;
